File: tests/07_phases_ompss.dg/fortran/success_taskyield_01.f90
! <testinfo>
! test_generator=(config/mercurium-ompss config/mercurium-ompss-v2)
! test_compile_fail_nanos6_mercurium=yes
! test_compile_fail_nanos6_imfc=yes
! </testinfo>

PROGRAM P
    IMPLICIT NONE

    !$OMP TASK
        !$OMP TASKYIELD
    !$OMP END TASK

    !$OMP TASKWAIT
END PROGRAM P

File: tests/07_phases_omp.dg/iomp/c/success_taskyield_01.c
/*
<testinfo>
test_generator=config/mercurium-iomp
</testinfo>
*/

/*This tests taskyield*/

#include <assert.h>
#include <unistd.h>
#include <omp.h>

omp_lock_t lck1;

int main(void) {
	omp_init_lock(&lck1);
	size_t sum = 0;
	#pragma omp parallel
	#pragma omp single
	{
		for (size_t i = 0; i < 100000; ++i) {
			#pragma omp task
			{
				while (!omp_test_lock(&lck1)) {
					#pragma omp taskyield
				}
				++sum;
				omp_unset_lock(&lck1);
			}
		}
	}
	omp_destroy_lock(&lck1);
	assert(sum == 100000);
}

